pkg/testworkflows/testworkflowexecutor/scheduler.go

```diff
@@ -117,7 +117,7 @@ func NewScheduler(
 
 func (s *scheduler) insert(ctx context.Context, execution *testkube.TestWorkflowExecution) error {
 	err := retry(SaveResultRetryMaxAttempts, SaveResultRetryBaseDelay, func() error {
-		err := s.resultsRepository.Insert(ctx, *execution)
+		err := s.resultsRepository.Insert(ctx, *execution.Clone())
```

Comment:
This should rather be cloned inside of repository. Otherwise the `Insert()`/`Update()` methods in the repository still have side effect.

Reply:
agree - better!